src/navigation/underline-nav.scss

```diff
@@ -1,18 +1,22 @@
+$nav-height: $spacer-3 * 3 !default; // 48px
+
 .UnderlineNav {
   display: flex;
-  overflow-x: auto;
```

Comment:
Removing `overflow-x: auto;` would make the items overflow when not enough space.

Hmmm... I guess if we add it back we would run into the same problem of the [outline being cut off](https://github.com/primer/css/pull/1744#issuecomment-976116586):

![Screen Shot 2021-11-24 at 14 09 14](https://user-images.githubusercontent.com/378023/143178695-23e60cc8-0ba7-4eed-87f5-5acf052d5f4e.png)

So maybe in this case we can make the focus outline appear as "inset" with `outline-offset: -2px;`.

Reply:
Ah, I forgot this overflows on small screens. Adding it back looks okay to me without adjusting the offset (still an improvement from no outline at all)

![image](https://user-images.githubusercontent.com/18661030/143286118-4687fa83-0f04-4e80-a564-5f515169910b.png)

The overflow state of this component could probably be improved, but I think Vinicius has some upcoming work planned to address look & feel.

Reply:
The cut off outline still looks a bit un-polished to me:

![Screen Shot 2021-11-25 at 11 11 54](https://user-images.githubusercontent.com/378023/143366111-49246197-2751-40e6-b1af-52f0a2e86556.png)

But should also fine if we improve it in a follow-up PR. Because yeah, the current focus state (gray underline) is barely visible. 🔍  I remember it having much more contrast, but maybe that changed with all the color modes updates.

For the repo nav, the items overflow into a dropdown menu:

![Screen Shot 2021-11-25 at 11 23 23](https://user-images.githubusercontent.com/378023/143366738-b14fdf00-a0c4-4625-90c7-542c177a131a.png)

So maybe that could be the long term plan for all `UnderlineNav`s? /cc @vdepizzol Maybe as PVC since it needs JS.

Reply:
Thanks @simurai! I didn't notice the focus cutting off like that in the `overflow` container. I adjusted it to `-2px` offset to compensate, and also added scroll snap behavior to hopefully improve the mobile experience a bit. It will still break over an item at the end unfortunately, but perhaps help prevent items getting cut in half a bit.